contracts/abstract/JBPayoutRedemptionPaymentTerminal.sol

```diff
@@ -860,15 +882,23 @@ abstract contract JBPayoutRedemptionPaymentTerminal is
               _netPayoutAmount = _payoutAmount;
               // This distribution is eligible for a fee since the funds are leaving this contract and the terminal isn't listed as feeless.
             else {
-              _netPayoutAmount = fee == 0 || _feeDiscount == JBConstants.MAX_FEE_DISCOUNT
+              _netPayoutAmount = _feeDiscount == JBConstants.MAX_FEE_DISCOUNT
                 ? _payoutAmount
-                : _payoutAmount - _getFeeAmount(_payoutAmount, _feeDiscount);
+                : _payoutAmount - _feeAmount(_payoutAmount, _feeDiscount);
 
               feeEligibleDistributionAmount += _payoutAmount;
             }
 
+            // Trigger any inherited pre-transfer logic.
             _beforeTransferTo(address(_terminal), _netPayoutAmount);
 
+            // Get a reference to the destination terminal's decimals.
+            uint256 _decimals = _terminal.decimals();
+
+            // If the destination terminal uses a different number of decimals than this terminal, adjust the sent amount accordingly.
+            if (_decimals != decimals)
+              _amount = JBFixedPointNumber.adjustDecimals(_amount, decimals, _decimals);
```

Comment:
Shouldn't this be
_netPayoutAmount = JBFixedPointNumber.adjustDecimals(_netPayoutAmount, decimals, _decimals);

Reply:
pushed up a solution that i think makes sense. The terminal will allow transfer given its set decimals, and set the transfer amount given the receiver's set decimals. its up to the receiving terminal to figure it out.
this is edge edge edge case, any properly setup terminal with the same token will have the same decimals.